ziti/collection.go

```diff
@@ -0,0 +1,143 @@
+/*
+	Copyright 2019 NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package ziti
+
+import (
+	"github.com/michaelquigley/pfxlog"
+	cmap "github.com/orcaman/concurrent-map/v2"
+	"os"
+	"strings"
+)
+
+// An SdkCollection allows Context instances to be instantiated and maintained as a group. Useful in scenarios
+// where multiple Context instances are managed together. Instead of using ziti.NewContext() like functions, use
+// the function provided on this type to automatically have contexts added as they are created. If ConfigTypes
+// is set, they will be automatically added to any instantiated Context through `New*` functions.
+type SdkCollection struct {
```

Comment:
SdkCollection? or ContextCollection? Why "SDK" vs "Context"?

Reply:
Length, but I can make it CtxCollection.